core/templates/dev/head/pages/collection_editor/CollectionEditorNavbarDirective.js

```diff
@@ -21,27 +21,35 @@ oppia.directive('collectionEditorNavbar', [function() {
     restrict: 'E',
     templateUrl: 'inline/collection_editor_navbar_directive',
     controller: [
-      '$scope', '$modal', 'alertsService', 'routerService', 'UndoRedoService',
-      'CollectionEditorStateService', 'CollectionValidationService',
-      'CollectionRightsBackendApiService',
-      'EditableCollectionBackendApiService',
+      '$scope', '$log', '$modal', '$rootScope', 'alertsService',
+      'routerService', 'UndoRedoService', 'CollectionEditorStateService',
+      'CollectionRightsBackendApiService', 'CollectionRightsObjectFactory',
+      'CollectionValidationService', 'EditableCollectionBackendApiService',
       'EVENT_COLLECTION_INITIALIZED', 'EVENT_COLLECTION_REINITIALIZED',
       'EVENT_UNDO_REDO_SERVICE_CHANGE_APPLIED',
       function(
-          $scope, $modal, alertsService, routerService, UndoRedoService,
-          CollectionEditorStateService, CollectionValidationService,
-          CollectionRightsBackendApiService,
-          EditableCollectionBackendApiService,
+          $scope, $log, $modal, $rootScope, alertsService,
+          routerService, UndoRedoService, CollectionEditorStateService,
+          CollectionRightsBackendApiService, CollectionRightsObjectFactory,
+          CollectionValidationService, EditableCollectionBackendApiService,
           EVENT_COLLECTION_INITIALIZED, EVENT_COLLECTION_REINITIALIZED,
           EVENT_UNDO_REDO_SERVICE_CHANGE_APPLIED) {
+        $scope.collectionRightsObject = null;
+        $scope.rightsHasLoaded = false;
```

Comment:
You could introduce a method which makes the check more readable, without the duplication of data:

```javascript
$scope.isCollectionsRightsObjectLoaded = function() {
  return $scope.collectionsRightsObject !== null;
};
```